### PR TITLE
[DC-711] Add request access for and modal

### PR DESCRIPTION
**Pull request by s-rubenstein**
### What
- Submits a request with the following body:
{
    "name": "test name",
    "researchPurposeStatement": "test research\n",
    "datasetRequest": {
        "cohorts": [
            {
                "name": "test",
                "criteriaGroups": [
                    {
                        "name": "Group 1",
                        "criteria": [
                            {
                                "kind": "domain",
                                "domainOption": {
                                    "kind": "domain",
                                    "id": 12,
                                    "category": "Observation",
                                    "conceptCount": 12300,
                                    "participantCount": 23223,
                                    "values": [
                                        "Blood Pressure",
                                        "Weight",
                                        "Height"
                                    ]
                                },
                                "name": "Blood Pressure",
                                "id": 1,
                                "count": 100
                            }
                        ],
                        "mustMeet": true,
                        "meetAll": false,
                        "count": 0
                    }
                ]
            }
        ],
        "conceptSets": [
            {
                "name": "Demographics",
                "featureValueGroupName": "Person"
            }
        ],
        "valuesSets": [
            {
                "domain": "Person",
                "values": [
                    {
                        "name": "demographics column 1"
                    },
                    {
                        "name": "demographics column 2"
                    }
                ]
            }
        ]
    }
}

Screenshot 2023-07-05 at 10 36 47 AM
Screenshot 2023-07-05 at 10 40 49 AM

### CODE REVIEW

**Review by pshapiro4broad — APPROVED** [review]
Looks good!

File: src/pages/library/datasetBuilder/DatasetBuilder.ts [hunk 1]
```diff
@@ -382,6 +385,86 @@ export const ValuesSelector = ({
...
+            onDismiss();
+          },
+        },
+        ['Request access']
+      ),
+    },
+    [
+      div([
+        div([
+          "A request of the dataset created will be generated and may take up to 72 hours for approval. Once approved you'll be notified by email. We'll send you a copy of this request",
```
Comment (okotsopoulos) [inline src/pages/library/datasetBuilder/DatasetBuilder.ts:436]: Curious about where the 72 hours comes from -- might this make more sense as "3 business days"?

**Review by okotsopoulos — APPROVED** [review]
Looks good!

File: src/pages/library/datasetBuilder/DatasetBuilder.ts (hunk 1 above)
Comment (s-rubenstein) [reply src/pages/library/datasetBuilder/DatasetBuilder.ts:436]: 72 hours came from the mocks, to be honest I am not super sure where it comes from. I can forward that question on to Lou when he is back.

Comment (okotsopoulos) [reply src/pages/library/datasetBuilder/DatasetBuilder.ts:436]: Sounds good! Not a blocker to merge.

**Pull request merged**